packages/story-editor/src/components/mediaRecording/provider.js

```diff
@@ -143,12 +143,23 @@ function MediaRecordingProvider({ children }) {
     // granted permission, so it's not easy to detect.
     // TODO: Figure out how to retry without microphone if possible.
     mediaStreamConstraints: {
-      audio: audioInput && hasAudio ? { deviceId: audioInput } : true,
-      video: videoInput && hasVideo ? { deviceId: videoInput } : true,
+      audio: hasAudio ? (audioInput ? { deviceId: audioInput } : true) : false,
+      video: hasVideo ? (videoInput ? { deviceId: videoInput } : true) : false,
     },
     onStop,
   });
 
+  useEffect(() => {
+    if (
+      error?.name === 'NotFoundError' ||
+      error?.name === 'NotReadableError' ||
+      error?.name === 'OverConstrainedError'
+    ) {
+      setHasVideo(false);
```

Comment:
Q: Should this influence the "Enable Video" quick-action as well? Can't test it unfortunately but looks like the button would still allow turning video on again (even though I suppose it'd give the error again and thus basically do nothing). Thoughts?

Reply:
In the next line we do `setVideoInput(null);` which hides the enable/disable video icon.